src/App.tsx

```diff
@@ -8,53 +9,115 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsersFromServer } from './api/users';
+import { Post } from './types/Post';
+import { getPostsFromServer } from './api/posts';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [activePostId, setActivePostId] = useState<number | null>(null)
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsersFromServer()
+      .then(setUsers)
+      .catch(() => {});
+  }, []);
```

Comment:
Consider adding error handling logic in the catch block of this useEffect hook. According to the checklist, you should use `try {..} catch` for error handling .